### ISSUE TITLE
TypeScript syntax error with inline fragments

### ISSUE THREAD

**Issue by epeli** [dotansimha/graphql-code-generator#1432]
**Describe the bug**

When inline fragments are used in unions with common `__typename` invalid TypeScript code is generated.

Ex.

```graphql
query TopMenu {
    menuItems(where: { location: TOP_MENU }) {
        edges {
            node {
                connectedObject {
                    __typename
                    ... on Page {
                        title
                    }
                    ... on Post {
                        title
                    }
                }
            }
        }
    }
}
```

with config

```yaml
generates:
    types.ts:
        - typescript-client:
              noNamespaces: true
```

Generates 

```ts
  export type TopMenuConnectedObject = {
    __typename: TopMenuPageInlineFragment["__typename"] | TopMenuPostInlineFragment["__typename"];
  } (TopMenuPageInlineFragment | TopMenuPostInlineFragment
```

Which is not syntactically valid TypeScript.

Error from prettier:

![image](https://user-images.githubusercontent.com/225712/54020172-c9052180-4195-11e9-8dd7-cce2a1c88c69.png)

**To Reproduce**

I've created a [repository](https://github.com/epeli/graphql-codegen-typescript-bug) demonstrating this:

```
git clone https://github.com/epeli/graphql-codegen-typescript-bug
cd graphql-codegen-typescript-bug/
npm ci
npm run codegen
npm test
```

**Expected behavior**

No really sure what should be generated here...

**Schema/Documents**

It's all in the repository

**Environment:**
 - OS: macos
 - Codegen: 0.18.0
 - Node: v10.15.1

**Comment by dotansimha** [comment]
Thanks @epeli 
We are currently working on a big refactor that will solve all these mis-matches and issues (https://github.com/dotansimha/graphql-code-generator/pull/1353). 
We hope to release a beta version in a day or two.

**Comment by dotansimha** [comment]
Fixed in the recent refactor :)

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍

**Comment by epeli** [comment]
Yay, thank you!
